File: TimeToToastApp/src/app/share/dtos/gift/gift-dto.ts
import { EventDto } from "../event/event-dto";
import { GiftContributionDto } from "../giftContribution/gift-contribution-dto";

export interface GiftDto {
    idGift:number,
    name: string,
    url:string,
    photo:string,
    wanted:number,
    price:number,
    categorie:string,
    isPaid:boolean,
    event:EventDto,
    contributions:GiftContributionDto[]
}

File: TimeToToast_Spring/src/main/java/com/maddoxgraham/TimeToToast/DTOs/GiftDto.java
package com.maddoxgraham.TimeToToast.DTOs;

import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class GiftDto {
private Long idGift;
    private String name;
    private String url;
    private String photo;
    private Long wanted;
    private Double price;

    private String categorie;
    private boolean isPaid;
    private Long event;
}
